atomic_reactor/plugins/post_koji_import.py

```diff
@@ -613,20 +564,126 @@ def _filesystem_koji_task_id(self) -> Optional[int]:
+    ) -> None:
+        wf_data = self.workflow.data
+        operator_manifests_path = wf_data.postbuild_results.get(
+            PLUGIN_EXPORT_OPERATOR_MANIFESTS_KEY
+        )
+        if not operator_manifests_path:
+            return
+
+        metadata = get_output_metadata(operator_manifests_path,
+                                       OPERATOR_MANIFESTS_ARCHIVE)
+        metadata['buildroot_id'] = buildroot_id
+        add_type_info(metadata, KOJI_BTYPE_OPERATOR_MANIFESTS)
+        outputs_metadata.append(metadata)
+
+        wf_data.koji_upload_files.append({
+            'local_filename': operator_manifests_path,
+            'dest_filename': OPERATOR_MANIFESTS_ARCHIVE,
+        })
 
+    def _collect_maven_metadata(
+            self, buildroot_id: str, outputs_metadata: List[Dict[str, Any]],
+    ) -> None:
+        wf_data = self.workflow.data
+        result = wf_data.postbuild_results.get(PLUGIN_MAVEN_URL_SOURCES_METADATA_KEY) or {}
+        for remote_source_file in result.get('remote_source_files', []):
+            metadata = remote_source_file['metadata']
+            metadata['buildroot_id'] = buildroot_id
+            outputs_metadata.append(metadata)
+            wf_data.koji_upload_files.append({
+                'local_filename': remote_source_file['file'],
+                'dest_filename': metadata['filename'],
+            })
+
+    def get_output(self, buildroot_id: str) -> List[Dict[str, Any]]:
+        """Assemble outputs specific to a binary build.
+
+        The corresponding files to be uploaded are also recorded for later
+        upload.
+
+        :param str buildroot_id: for binary build, this argument is ignored.
+            Instead, use the buildroot id which is already set in the build
+            metadata.
         :return: list, containing dicts of partial metadata
         """
-        outputs = []
-        output_file = None
+        wf_data = self.workflow.data
 
-        for platform, instance in self._iter_build_metadata_outputs():
-            instance['buildroot_id'] = '{}-{}'.format(platform, instance['buildroot_id'])
-            outputs.append(instance)
+        result = wf_data.prebuild_results.get(PLUGIN_FETCH_MAVEN_KEY) or {}
+        maven_components = result.get('components', [])
 
-        return outputs, output_file
+        platform: str
+        output: Dict[str, Any]  # an output metadata of the build
+        outputs: List[Dict[str, Any]] = []
+
+        for platform, output in self._iter_build_metadata_outputs():
+            buildroot_id = output['buildroot_id']
+            output['buildroot_id'] = f'{platform}-{buildroot_id}'
+            if maven_components and output['type'] == 'docker-image':
+                # add maven components alongside RPM components
+                output['components'] += maven_components
+            outputs.append(output)
+
+        buildroot_id = outputs[0]['buildroot_id']
```

Comment:
This method first sets the buildroot_id of all per-platform outputs to `{platform}-{buildroot_id}`, then chooses an arbitrary `{platform}-{buildroot_id}` to use for the rest of the outputs. Shouldn't those remaining outputs use a platform-less id?

Reply:
> buildroot_id: The id of the buildroot used to create this file. Must match an entry in the buildroots list.[1]

The Koji doc tells this.

[1] https://docs.pagure.org/koji/content_generator_metadata/#output

Reply:
I see, and this _is_ the same as what we had in OSBS 1, right?

Reply:
I think so. In OSBS 1, the first platform buildroot id is used to collect the artifacts. In this pr, the first output has the first platform buildroot id.